ansible/files/vmware_api.py

```diff
@@ -239,6 +227,52 @@ def get_vm_clone_spec(config_spec, customization_spec, relocate_spec):
     return clone_spec
 
 
+def get_vm_storage_spec(name, folder, pod_selection_spec, vm, clone_spec, operation_type):
+    storage_spec = vim.storageDrs.StoragePlacementSpec()
+    storage_spec.cloneName = name
+    storage_spec.folder = folder
+    storage_spec.podSelectionSpec = pod_selection_spec
+    storage_spec.vm = vm
+    storage_spec.cloneSpec = clone_spec
+    storage_spec.type = operation_type
+    return storage_spec
+
+def get_vm_pod_selection_spec(api, storage_pod_name):
+    storage_pod = _get_storage_pod(api, storage_pod_name)
+    pod_selection_spec = vim.storageDrs.PodSelectionSpec()
+    pod_selection_spec.storagePod = storage_pod
+    return pod_selection_spec
+
+
+def _get_storage_pod(api, storage_pod_name):
+    storage_pod = api.get_vc_object(vim.StoragePod, storage_pod_name)
+    return storage_pod
+
+
+def get_apply_storage_recommendation_task(api, storage_spec):
```

Comment:
Maybe `clone_template_to_datastore_cluster` would be a better name for this function.